com.github.elliegames.sanctify-game.yml

```diff
@@ -14,36 +12,105 @@ finish-args:
-      - install -D data/icons/48/com.github.elliegames.sanctify-game.svg /app/share/icons/hicolor/48x48/apps/com.github.elliegames.sanctify-game.svg
-      - install -D data/icons/48/com.github.elliegames.sanctify-game.svg /app/share/icons/hicolor/48x48@2/apps/com.github.elliegames.sanctify-game.svg
-      - install -D data/icons/64/com.github.elliegames.sanctify-game.svg /app/share/icons/hicolor/64x64/apps/com.github.elliegames.sanctify-game.svg
-      - install -D data/icons/64/com.github.elliegames.sanctify-game.svg /app/share/icons/hicolor/64x64@2/apps/com.github.elliegames.sanctify-game.svg
-      - install -D data/icons/128/com.github.elliegames.sanctify-game.svg /app/share/icons/hicolor/128x128/apps/com.github.elliegames.sanctify-game.svg
-      - install -D data/icons/128/com.github.elliegames.sanctify-game.svg /app/share/icons/hicolor/128x128@2/apps/com.github.elliegames.sanctify-game.svg
-      - install -D data/icons/symbolic/com.github.elliegames.sanctify-game.svg /app/share/icons/hicolor/symbolic/apps/com.github.elliegames.sanctify-game.svg
-      - install -D data/com.github.elliegames.sanctify-game.desktop /app/share/applications/com.github.elliegames.sanctify-game.desktop
-      - install -D data/com.github.elliegames.sanctify-game.appdata.xml /app/share/appdata/com.github.elliegames.sanctify-game.appdata.xml
+      - pip3 install --no-index --no-build-isolation --prefix=/app .
 
+  - name: godot
+    buildsystem: simple
+    cleanup:
+      - /bin/godot-tools
     sources:
       - type: archive
-        url: https://downloads.tuxfamily.org/godotengine/4.3/Godot_v4.3-stable_linux.x86_64.zip
-        sha256: 7de56444b130b10af84d19c7e0cf63cf9e9937ee4ba94364c3b7dd114253ca21
-     
+        url: https://github.com/godotengine/godot/releases/download/4.3-stable/godot-4.3-stable.tar.xz
+        sha256: 751e55bfad8e04b846f9cf7b6eb80e058986a2cb1b103fc0fe6a4d8526a20e56
+
+    build-commands:
+      # Tools, to export the PCK file from source at build time
+      - python3 /app/bin/scons $SCONS_FLAGS $SCONS_FLAGS_EXTRA tools=yes target=editor -j "$FLATPAK_BUILDER_N_JOBS"
+      - install -Dm755 bin/godot.linuxbsd.editor.* /app/bin/godot-tools
+
+      # Runner, to be shipped in the Flatpak as the binary
+      - python3 /app/bin/scons $SCONS_FLAGS $SCONS_FLAGS_EXTRA tools=no target=template_release -j "$FLATPAK_BUILDER_N_JOBS"
+      - install -Dm755 bin/godot.linuxbsd.template_release.* /app/bin/${FLATPAK_ID}
+
+      # Export templates, used at build-time to export the PCK file
+      - touch _sc_
+      - mkdir -p ./editor_data/export_templates
+      - mv bin/ editor_data/export_templates/4.3.stable
+
+  - name: sanctify-game
+    buildsystem: simple
+    sources:
       - type: dir
         path: .
+
+    build-commands:
+      # Export the PCK file and install it alongside the Godot Engine runner
+      - godot-tools --path sanctify --headless --verbose --export-pack "Linux" ${FLATPAK_ID}.pck
```

Comment:
If you `--export` here instead of `--export-pack`, and pass `/app/bin/${FLATPAK_ID}` as the dest, you can export the .pck and the correctly named executable at the same time, meaning you don't have to export the executable as part of the godot module above, allowing `cleanup: [*]`, which maybe makes that module a bit cleaner to drop into other manifests.

Reply:
Doesn't `--export` require export templates? Or is that produced as part of the godot build step?

Reply:
Yeah, the previous module sets up the export template.

I'm using `--export` in my PR and it's tested and working.

Reply:
@davidmhewitt ah yeah, I was coming from the Flathub world where a BaseApp provides the runner, so you only provide the PCK file. But of course if you're building it all from source anyway… this makes sense.